checker/src/main/java/org/checkerframework/checker/index/upperbound/UpperBoundVisitor.java

```diff
@@ -308,6 +410,37 @@ private boolean relaxedCommonAssignmentCheck(
         return true;
     }
 
+    /* Returns the new value of the left hand side after processing the arrays named in the lhs.
+     * Iff varLtlQual includes LTL(lhsSeq),
+     * lhsSeq has HSS, and expQual includes LTL(a, -from), then the LTL(lhsSeq) will be removed from varLtlQual
+     */
+    private UBQualifier processSubsequenceForLHS(LessThanLengthOf varLtlQual, UBQualifier expQual) {
+        UBQualifier newLHS = varLtlQual;
+        for (String lhsSeq : varLtlQual.getSequences()) {
+            // check is lhsSeq is an actual LTL
+            if (varLtlQual.hasSequenceWithOffset(lhsSeq, 0)) {
+
+                Receiver rec =
+                        getReceiverFromJavaExpressionString(lhsSeq, atypeFactory, getCurrentPath());
+                FlowExpressionContext context = Subsequence.getContextFromReceiver(rec, checker);
+                Subsequence subSeq =
+                        Subsequence.getSubsequenceFromReceiver(
+                                rec, atypeFactory, getCurrentPath(), context);
+
+                if (subSeq != null) {
+                    String from = subSeq.from;
+                    String a = subSeq.array;
+
+                    if (expQual.hasSequenceWithOffset(a, negateString(from, context))) {
+                        // this cast is safe because LTLs cannot contain duplicates
+                        newLHS = ((LessThanLengthOf) newLHS).removeOffset(lhsSeq, 0);
```

Comment:
`newLHS` might be re-assigned twice, if `varLtlQual` has two sequences with `@HasSubseq`.  I'm not really worried about that happening, but you should note that here and possible just return newLHS here.

Reply:
That shouldn't be a problem, since it's updating from `newLHS` each time. I'll add a note here, though.

Reply:
Oh, yes, I see that now.